### DISCUSSION TITLE
How to modify slurm template to run inside singularity container

### DISCUSSION THREAD

**Discussion by mattwarkentin** [mschubert/clustermq#252] [Q&A] [answered]
Hi @mschubert,

I was just wondering how I should modify my slurm template file in order to run the `R` worker processes inside a singularity container. Does the below look correct? Basically just taking the entire last line in the default template file and executing it with `singularity exec container.sif ...`.
```bash
#!/usr/bin/bash

#SBATCH ...
#SBATCH ...

module load singularity

ulimit -v $(( 1024 * {{ memory | 4096 }} ))
singularity exec container.sif CMQ_AUTH={{ auth }} R --no-save --no-restore -e 'clustermq:::worker("{{ master }}")'
```

**Comment by mschubert** [comment]
That's how I think it should work!

Note that the container needs to be able to access the network on the host system.

**Comment by mattwarkentin** [comment]
Okay, thanks! I'm about to test it out so I will report back with how it goes.

**Comment by mattwarkentin** [comment, marked as answer]
Okay so it works well, but only after I removed the `CMQ_AUTH={{ auth }}` from the final line. Otherwise it produces this error:
```
/.singularity.d/actions/exec: 9: exec: CMQ_AUTH=ztbsf: not found
```
